app/src/main/java/io/homeassistant/companion/android/sensors/SensorDetailFragment.kt

```diff
@@ -267,61 +275,23 @@ class SensorDetailFragment(
-                        pref.title = setting.name
-                        pref.entries = packageName.toTypedArray()
-                        pref.entryValues = packageName.toTypedArray()
-                        pref.dialogTitle = setting.name
-                        pref.isIconSpaceReserved = false
-                        pref.isSingleLineTitle = false
-                        pref.setOnPreferenceChangeListener { _, newValue ->
-                            sensorDao.add(
-                                    Setting(
-                                            basicSensor.id,
-                                            setting.name,
-                                            newValue.toString().replace("[", "").replace("]", ""),
-                                            "list-apps"
-                                    )
-                            )
-                            sensorManager.requestSensorUpdate(requireContext())
-                            return@setOnPreferenceChangeListener true
-                        }
-                        if (pref.values != null)
-                            pref.summary = pref.values.toString()
-                        else
-                            pref.summary = setting.value
+
+                        val pref = createListPreference(key, setting, sensorDao, packageName)
                         if (!it.contains(pref))
                             it.addPreference(pref)
                     } else if (setting.valueType == "list-bluetooth") {
                         val btDevices = BluetoothUtils.getBluetoothDevices(requireContext()).map { b -> b.name }
 
-                        val pref = findPreference(key)
-                                ?: MultiSelectListPreference(requireContext())
-                        pref.key = key
-                        pref.title = setting.name
-                        pref.entries = btDevices.toTypedArray()
-                        pref.entryValues = btDevices.toTypedArray()
-                        pref.dialogTitle = setting.name
-                        pref.isIconSpaceReserved = false
-                        pref.isSingleLineTitle = false
-                        pref.setOnPreferenceChangeListener { _, newValue ->
-                            sensorDao.add(
-                                    Setting(
-                                            basicSensor.id,
-                                            setting.name,
-                                            newValue.toString().replace("[", "").replace("]", ""),
-                                            "list-bluetooth"
-                                    )
-                            )
-                            sensorManager.requestSensorUpdate(requireContext())
-                            return@setOnPreferenceChangeListener true
+                        val pref = createListPreference(key, setting, sensorDao, btDevices)
+                        if (!it.contains(pref))
```

Comment:
"If the condition of an if or when statement is multiline, always use curly braces around the body of the statement." https://kotlinlang.org/docs/coding-conventions.html#control-flow-statements  Kotlin coding-conventions

Reply:
Thanks for the tip. But do I miss something here? The if isn't multiline. Personally I would prefer the if as an one liner or at least with braces around. I would change this one to a one liner. 
But if the Kotlin conventions recommend a different way, I would adapt my preferences to that.